src/API/apiController.tsx

```diff
@@ -1,10 +1,34 @@
+import { apiControllerStrings } from '../constants/strings';
+import * as reducer from '../features/vod/reducers/vodCategories.reducer';
+import { VODCategory } from '../constants/types';
+
 export const getMenu = async () => {
-  const url = 'https://sp1liv.maariv.co.il/wp-json/sport1/v1/navigation';
+  return apiCall(apiControllerStrings.mainMenu);
+};
+
+export const getVODMenu = () => {
+  let categories: VODCategory[] = [];
+  apiCall(apiControllerStrings.vodMenu)
+    .then(result => {
+      if (result) {
+        result.json().then(json => {
+          categories = json.categories;
+          reducer.setVODCategories(categories);
+        });
+      }
+    })
+    .catch(() => {
+      reducer.setVODCategories(null);
```

Comment:
what about handling the error?

Reply:
Because we don't have the list of Error codes to handle each error, I pass `null` to the reducer that assigns `null` to `categories` property in the store's state, and that's how I know there has been an error during the API call. Triggering it on UI by a constant string.